agents/Group23/mcts.py

```diff
@@ -15,58 +16,52 @@ def __init__(self, colour: Colour, max_simulation_length: float = 2.5, custom_tr
         self.colour = colour  # Agent's colour
         self.max_simulation_length = max_simulation_length  # Length of a MCTS search in seconds
 
-    def _get_visit_count_distribution(self, node: TreeNode) -> list[list[int]]:
-        """Returns the visit count distribution for the children of the given node.
+    def run_simulation_with_process(self, root: TreeNode, colour: Colour,start_time: float) -> int:
+        """Runs a simulation with a new process."""
 
-        Args:
-            node (TreeNode): current node
+        iterations = 0
 
-        Returns:
-            list[list[int]]: visit count distribution
-        """
-        distribution_board = [[0 for _ in range(11)] for _ in range(11)]
-        self._count_visits_DFS(node, distribution_board)
-
-        # softmax normalization
-        total_visits = sum(sum(row) for row in distribution_board)
-        for i in range(11):
-            for j in range(11):
-                distribution_board[i][j] /= total_visits
-        return distribution_board
-    
-    def _count_visits_DFS(self, node: TreeNode, distribution_board: list[list[int]]):
-        """Counts the visits for the children of the given node.
+        while time.time() - start_time < self.max_simulation_length:
+            node = self._select(root)
+            result = self._simulate(node.board, colour)
+            self._backpropagate(node, result)
+            iterations += 1
 
-        Args:
-            node (TreeNode): current node
-            distribution_board (list[list[int]]): visit count distribution passed by reference
-        """
-        for child in node.children:
-            x, y = child.move.x, child.move.y
-            distribution_board[x][y] += child.visits
-            self._count_visits_DFS(child, distribution_board)
+        # delete second children
+        for child in root.children:
```

Comment:
If we aren't using Alpha-Zero after all, is it worth us removing the methods for count_visits and the distribution board, since it won't be used anywhere? Removes a now-redundant computation

Reply:
Yeah, I had another look at alpha zero earlier and I think MCTS is the way to go

Reply:
Perhaps just remove these lines then :)

Reply:
I think all the code related to that has been removed, so good to go :)